[React18] Migrate test suites to account for testing library upgrades security-threat-hunting-explore

eokoneyo opened the pull request:
This PR migrates test suites that use `renderHook` from the library `@testing-library/react-hooks` to adopt the equivalent and replacement of `renderHook` from the export that is now available from
`@testing-library/react`. This work is required for the planned migration to react18. 

##  Context

In this PR, usages of `waitForNextUpdate` that previously could have been destructured from `renderHook` are now been replaced with `waitFor` exported from `@testing-library/react`, furthermore `waitFor` 
that would also have been destructured from the same renderHook result is now been replaced with `waitFor` from the export of `@testing-library/react`.

***Why is `waitFor` a sufficient enough replacement for `waitForNextUpdate`, and better for testing values subject to async computations?***

WaitFor will retry the provided callback if an error is returned, till the configured timeout elapses. By default the retry interval is `50ms` with a timeout value of `1000ms` that 
effectively translates to at least 20 retries for assertions placed within waitFor. See https://testing-library.com/docs/dom-testing-library/api-async/#waitfor for more information.
This however means that for person's writing tests, said person has to be explicit about expectations that describe the internal state of the hook being tested. 
This implies checking for instance when a react query hook is being rendered, there's an assertion that said hook isn't loading anymore.

In this PR you'd notice that this pattern has been adopted, with most existing assertions following an invocation of `waitForNextUpdate` being placed within a `waitFor` 
invocation. In some cases the replacement is simply a `waitFor(() => new Promise((resolve) => resolve(null)))` (many thanks to @kapral18, for point out exactly why this works),
 where this suffices the assertions that follow aren't placed within a waitFor so this PR doesn't get larger than it needs to be.

It's also worth pointing out this PR might also contain changes to test and application code to improve said existing test.

### What to do next?
1. Review the changes in this PR.
2. If you think the changes are correct, approve the PR.

## Any questions?
If you have any questions or need help with this PR, please leave comments in this PR.


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

eokoneyo committed:
[React18] Migrated test suites to accommodate changes to testing library owned by security-threat-hunting-explore

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

eokoneyo commented:
/ci

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/253190)
* Commit: d6225b6680c93dd061ee348cc9d2c361eeaef441

### Failed CI Steps
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/253190#01934f4f-561d-41e7-be27-0c1741f7d7ce)
* [Jest Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/253190#01934f4f-5622-4239-9d55-77a643c6ec44)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/253190#01934f4f-561d-41e7-be27-0c1741f7d7ce) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/253190/jobs/01934f4f-561d-41e7-be27-0c1741f7d7ce/artifacts/01934f78-ca76-4060-8462-c9c55b70399b) Jest Tests #<span></span>6 / Policy Advanced Settings section should expand and collapse section when button is clicked

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/51a84ebd1d8d2a8bbe9fc80216acc3e0b2d7ddb8) | [after](https://github.com/elastic/kibana/commit/d6225b6680c93dd061ee348cc9d2c361eeaef441) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 13.4MB | 13.4MB | +185.0B |



cc @eokoneyo

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253190","number":253190,"commit":"d6225b6680c93dd061ee348cc9d2c361eeaef441"}],"number":253190}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

eokoneyo commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into chore/testing-library-migration-for-security-threat-hunting-explore

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/253846)
* Commit: 46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dab45ef33a5af5332e71e60dc74c3af71ca1746c) | [after](https://github.com/elastic/kibana/commit/46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 13.3MB | 13.3MB | +185.0B |



### History
* :yellow_heart: [Build #253190](https://buildkite.com/elastic/kibana-pull-request/builds/253190) was flaky d6225b6680c93dd061ee348cc9d2c361eeaef441

cc @eokoneyo

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253846","number":253846,"commit":"46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253190","number":253190,"commit":"d6225b6680c93dd061ee348cc9d2c361eeaef441"}],"number":253846}
buildkite-pr-comment-->

kapral18 commented:
test

kapral18 commented on packages/kbn-cell-actions/src/context/cell_actions_context.test.tsx:
```diff
@@ -7,7 +7,7 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
```
test

eokoneyo commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into chore/testing-library-migration-for-security-threat-hunting-explore

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/254789)
* Commit: 322572af226bc6d99b6256772b168698eb6e854c

### Failed CI Steps
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/254789#01936cfe-7ff0-4b9c-b53b-7f2030188b94)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/254789#01936cfe-8016-4fad-b73a-6cb915c47b54)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/254789#01936cfe-8016-4fad-b73a-6cb915c47b54) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/254789/jobs/01936cfe-8016-4fad-b73a-6cb915c47b54/artifacts/01936d22-ee13-4307-af66-d1bbbe53f2be) FTR Configs #<span></span>12 / management Index patterns on aliases discover verify hits should be able to discover and verify no of hits for alias2
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/254789#01936cfe-7ff0-4b9c-b53b-7f2030188b94) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/254789/jobs/01936cfe-7ff0-4b9c-b53b-7f2030188b94/artifacts/01936d26-9bc2-45c7-a876-111729f48159) Jest Tests #<span></span>4 / StepDefinePackagePolicy default API response should display vars coming from package policy
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/254789#01936cfe-7ff0-4b9c-b53b-7f2030188b94) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/254789/jobs/01936cfe-7ff0-4b9c-b53b-7f2030188b94/artifacts/01936d26-9bc2-4761-9b59-d54144446790) Jest Tests #<span></span>4 / useActionStatus should report error on cancel upgrade failure

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2e418d4dab8a60b88bce66a92bd45eb03010ee05) | [after](https://github.com/elastic/kibana/commit/322572af226bc6d99b6256772b168698eb6e854c) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 13.4MB | 13.4MB | +185.0B |



### History
* :green_heart: [Build #253846](https://buildkite.com/elastic/kibana-pull-request/builds/253846) succeeded 46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014
* :yellow_heart: [Build #253190](https://buildkite.com/elastic/kibana-pull-request/builds/253190) was flaky d6225b6680c93dd061ee348cc9d2c361eeaef441

cc @eokoneyo

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254789","number":254789,"commit":"322572af226bc6d99b6256772b168698eb6e854c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253846","number":253846,"commit":"46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253190","number":253190,"commit":"d6225b6680c93dd061ee348cc9d2c361eeaef441"}],"number":254789}
buildkite-pr-comment-->

eokoneyo commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into chore/testing-library-migration-for-security-threat-hunting-explore

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/255883)
* Commit: c6ddf49478fdd448a26f75485aea2bb51f11f5df

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/397314c9524bac9ea3e13155e352a889e5cd8cbe) | [after](https://github.com/elastic/kibana/commit/c6ddf49478fdd448a26f75485aea2bb51f11f5df) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 14.6MB | 14.6MB | +222.0B |



### History
* :yellow_heart: [Build #254789](https://buildkite.com/elastic/kibana-pull-request/builds/254789) was flaky 322572af226bc6d99b6256772b168698eb6e854c
* :green_heart: [Build #253846](https://buildkite.com/elastic/kibana-pull-request/builds/253846) succeeded 46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014
* :yellow_heart: [Build #253190](https://buildkite.com/elastic/kibana-pull-request/builds/253190) was flaky d6225b6680c93dd061ee348cc9d2c361eeaef441

cc @eokoneyo

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/255883","number":255883,"commit":"c6ddf49478fdd448a26f75485aea2bb51f11f5df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254789","number":254789,"commit":"322572af226bc6d99b6256772b168698eb6e854c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253846","number":253846,"commit":"46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253190","number":253190,"commit":"d6225b6680c93dd061ee348cc9d2c361eeaef441"}],"number":255883}
buildkite-pr-comment-->

kapral18 commented:
@eokoneyo added a PR with corrections https://github.com/elastic/kibana/pull/202586

kapral18 committed:
test: refactor tests to use async waitFor and improve asynchronous ha… (#202586)

…ndling

Addresses https://github.com/elastic/kibana/pull/201142

- replace waitFor(() => new Promise(resolve => resolve(null))) calls
with waitFor() expectation blocks
- wrap remaining expectation blocks with waitFor() to avoid act warnings
- Rename test files from `.test.ts` to `.test.tsx` where necessary.
- Refactor components to use `FC` and `PropsWithChildren` for better
TypeScript support.
- Remove unnecessary `act` wrappers and streamline test setup functions.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kapral18 reviewed (approved):
LGTM

eokoneyo commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into chore/testing-library-migration-for-security-threat-hunting-explore

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/12163767051

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/256883)
* Commit: 5200f2fd9f27de385218fff893174b24d95b1faf

### Failed CI Steps
* [Jest Integration Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/256883#019391cd-99cb-440f-bae0-38507a5be08c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #255883](https://buildkite.com/elastic/kibana-pull-request/builds/255883) succeeded c6ddf49478fdd448a26f75485aea2bb51f11f5df
* :yellow_heart: [Build #254789](https://buildkite.com/elastic/kibana-pull-request/builds/254789) was flaky 322572af226bc6d99b6256772b168698eb6e854c
* :green_heart: [Build #253846](https://buildkite.com/elastic/kibana-pull-request/builds/253846) succeeded 46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014
* :yellow_heart: [Build #253190](https://buildkite.com/elastic/kibana-pull-request/builds/253190) was flaky d6225b6680c93dd061ee348cc9d2c361eeaef441

cc @eokoneyo

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/256883","number":256883,"commit":"5200f2fd9f27de385218fff893174b24d95b1faf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/255883","number":255883,"commit":"c6ddf49478fdd448a26f75485aea2bb51f11f5df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254789","number":254789,"commit":"322572af226bc6d99b6256772b168698eb6e854c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253846","number":253846,"commit":"46a07f118909e47f6fd6cbe93bc7bbf6fdd9b014"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/253190","number":253190,"commit":"d6225b6680c93dd061ee348cc9d2c361eeaef441"}],"number":256883}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/202978">](https://github.com/elastic/kibana/pull/202978)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
